### ISSUE TITLE
Consider not checking in autogenerated core/{Tensor.h,TensorMethods.h}

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#24931]
ATen/core/Tensor.h, ATen/core/TensorMethods.h are autogenerated and checked in. However, checking in autogenerated files is a recipe for merge conflicts; I've ran into 3 in the past two weeks. I think this is because git thinks it can merge two of these files from different branches together but they get merged together in a way that isn't consistent with how they get autogenerated.

Potential solutions:
- stop checking in the autogenerated files
- Have some sort of landing hook (I'm not sure if this is possible) where if the generated file doesn't match what is checked in, then we block the land.

cc @gchanan @ezyang

**Comment by gchanan** [comment]
This may be possible now because we don't actually need the Tensor class for the mobile build (we thought we did and we didn't build codegen for xplat).  But I'm not sure if that's the long-term plan.  Probably @ljk53 would know better.

**Comment by ezyang** [comment]
Yes let's stop checking it in. We'll need to macro our way to victory but it seems far preferable.

**Comment by gchanan** [comment]
I'm not sure why we need to macro our way to victory.

**Comment by ezyang** [comment]
My idea is that mobile would have a copy of Tensor.h but it just wouldn't have any methods on it.

**Comment by ljk53** [comment]
@gchanan @ezyang - just FYI I've already updated xplat/caffe2/BUCK to build aten & torch with all the codegen stuff. If the original concern to check in Tensor.h was due to codegen then it's not the case any more :)

**Comment by pearu** [comment]
I was hit by the same issue in PR #24937 .
For debugging it, here's an implementation of showing the diff of nonmatching files:
https://github.com/pytorch/pytorch/pull/24937/files#diff-76aa8cb3d0fad99c5f761d08cbcb4d19R372

**Comment by gchanan** [comment]
@ljk53 that's great!  Let me try throwing up a patch that gets rid of the src checkin.
